src/cascadia/TerminalSettingsEditor/ColorSchemes.cpp

```diff
@@ -2,99 +2,111 @@
-    {
-        m_colorSchemeModel.ColorScheme().BrightBlue(event.NewColor());
-    }
-    void ColorSchemes::Cyan_ColorChanged(ColorPicker const&, ColorChangedEventArgs const& event)
+    IObservableVector<hstring> ColorSchemes::ColorSchemeList()
     {
-        m_colorSchemeModel.ColorScheme().Cyan(event.NewColor());
+        return _ColorSchemeList;
     }
-    void ColorSchemes::BrightCyan_ColorChanged(ColorPicker const&, ColorChangedEventArgs const& event)
-    {
-        m_colorSchemeModel.ColorScheme().BrightCyan(event.NewColor());
-    }
-    void ColorSchemes::Green_ColorChanged(ColorPicker const&, ColorChangedEventArgs const& event)
-    {
-        m_colorSchemeModel.ColorScheme().Green(event.NewColor());
-    }
-    void ColorSchemes::BrightGreen_ColorChanged(ColorPicker const&, ColorChangedEventArgs const& event)
-    {
-        m_colorSchemeModel.ColorScheme().BrightGreen(event.NewColor());
-    }
-    void ColorSchemes::Purple_ColorChanged(ColorPicker const&, ColorChangedEventArgs const& event)
-    {
-        m_colorSchemeModel.ColorScheme().Purple(event.NewColor());
-    }
-    void ColorSchemes::BrightPurple_ColorChanged(ColorPicker const&, ColorChangedEventArgs const& event)
-    {
-        m_colorSchemeModel.ColorScheme().BrightPurple(event.NewColor());
-    }
-    void ColorSchemes::Red_ColorChanged(ColorPicker const&, ColorChangedEventArgs const& event)
+
+    void ColorSchemes::ColorSchemeSelectionChanged(IInspectable const& /*sender*/,
+                                                   SelectionChangedEventArgs const& args)
     {
-        m_colorSchemeModel.ColorScheme().Red(event.NewColor());
+        //  Update the color scheme this page is modifying
+        auto str = winrt::unbox_value<hstring>(args.AddedItems().GetAt(0));
+        auto colorScheme = MainPage::Settings().GlobalSettings().ColorSchemes().Lookup(str);
+        CurrentColorScheme(colorScheme);
+        _UpdateColorTable(colorScheme);
     }
-    void ColorSchemes::BrightRed_ColorChanged(ColorPicker const&, ColorChangedEventArgs const& event)
+
+    void ColorSchemes::_UpdateColorSchemeList()
     {
-        m_colorSchemeModel.ColorScheme().BrightRed(event.NewColor());
+        auto colorSchemeMap = MainPage::Settings().GlobalSettings().ColorSchemes();
+        for (const auto& pair : MainPage::Settings().GlobalSettings().ColorSchemes())
+        {
+            _ColorSchemeList.Append(pair.Key());
+        }
     }
-    void ColorSchemes::White_ColorChanged(ColorPicker const&, ColorChangedEventArgs const& event)
+
+    void ColorSchemes::ColorPickerChanged(IInspectable const& sender,
+                                          ColorChangedEventArgs const& /*args*/)
     {
-        m_colorSchemeModel.ColorScheme().White(event.NewColor());
+        if (auto picker = sender.try_as<ColorPicker>())
+        {
+            // TODO: Commented out for now because Tag currently won't bind to an index correctly.
```

Comment:
no bare todos, leave a work item ID

Reply:
I'm actually actively working on this on a separate branch, just wanted to comment this part out so that this PR doesn't get affected by a potentially buggy event handler.

Reply:
oh ok fine.